pkg/template/api/validation/validation.go

```diff
@@ -33,6 +40,11 @@ func ValidateProcessedTemplate(template *api.Template) fielderrors.ValidationErr
 func ValidateTemplate(template *api.Template) (errs fielderrors.ValidationErrorList) {
 	errs = validation.ValidateObjectMeta(&template.ObjectMeta, true, validation.ValidatePodName).Prefix("metadata")
 	errs = append(errs, validateTemplateBody(template)...)
+	fields := map[string]string{}
+	for i := range template.Parameters {
+		fields[template.Parameters[i].Name] = template.Parameters[i].Description
+	}
+	errs = append(errs, validation.ValidateAnnotations(fields, "description")...)
```

Comment:
it does not... and the key length is restricted to 63 character which will make @smarterclayton sad... I think I will have to copy the func from k8s and validate keys based on our regexp...